### ISSUE TITLE
Locale code has similar part as page name

### ISSUE THREAD

**Issue by acibilous** [victorgarciaesgi/nuxt-typed-router#99]
I have the project config:
```
export default defineNuxtConfig({
  devtools: { enabled: true },
  modules: [
    'nuxt-typed-router',
    '@nuxtjs/i18n'
  ],
  i18n: {
    defaultLocale: 'us',
    locales: ['us', 'ca'],
  },
})
```
and page with name `cart.vue`
As **ca**rt.vue is similar 'ca' locale code generation of this route definition will be broken.
<img width="1036" alt="image" src="https://github.com/victorgarciaesgi/nuxt-typed-router/assets/28527402/f164a4d6-5fba-4c59-9710-bef5867f071c">
This condition will return true and `} else if (route.name && !isLocaleRoute) {` will not be achieved
<img width="1010" alt="image" src="https://github.com/victorgarciaesgi/nuxt-typed-router/assets/28527402/af5c48cd-f11b-4299-814a-6fb936376920">

Expected result:
route for cart page is generating

**Comment by victorgarciaesgi** [comment]
Should probably make a big cleanup of the code, sorry you had to dig in that 😅

**Comment by victorgarciaesgi** [comment]
Fixed in `3.2.3`

**Comment by acibilous** [comment]
Thanks for fixing it!
